### PR TITLE
feat(dli): add new datasource to get list of spark templates

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:
add new datasource to get list of spark templates

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dli" TESTARGS="-run TestAccDataSourceDliSparkTemplates_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDataSourceDliSparkTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDliSparkTemplates_basic
=== PAUSE TestAccDataSourceDliSparkTemplates_basic
=== CONT  TestAccDataSourceDliSparkTemplates_basic
--- PASS: TestAccDataSourceDliSparkTemplates_basic (45.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       45.207s
```

### CODE REVIEW

File: docs/data-sources/dli_spark_templates.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Use this data source to get the list of the DLI spark templates.
+
+## Example Usage
+
+```hcl
+variable "template_id" {}
+data "huaweicloud_dli_spark_templates" "test" {
+  template_id = var.template_id
+}
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
variable "template_id" {}

data "huaweicloud_dli_spark_templates" "test" {
  template_id = var.template_id
}
```
using white line to split the difference blocks

File: docs/data-sources/dli_spark_templates.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `template_id` - (Optional, String) Specifies the ID of the spark template.
+
+* `name` - (Optional, String) Specifies the name of the spark template.
+
+* `group` - (Optional, String) Specifies the group of the spark template.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `name` - (Optional, String) Specifies the name of the spark template to be queried.

* `group` - (Optional, String) Specifies the group name to which the spark templates belong.
```

File: docs/data-sources/dli_spark_templates.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* `group` - The group of template.
+
+<a name="templates_body_struct"></a>
+The `body` block supports:
+
+* `app_parameters` - The input parameters of the main class, that is application parameters.
+
+* `name` - The Spark job name, the value contains a maximum of 128 characters.
+
+* `obs_bucket` - The OBS bucket for storing the Spark jobs.  Set this parameter when you need to save jobs.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `obs_bucket` - The OBS bucket for storing the Spark jobs.
```
schema definition ditto.

File: docs/data-sources/dli_spark_templates.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+* `auto_recovery` - The whether to enable the retry function.
+  If enabled, Spark jobs will be automatically retried after an exception occurs.
+  The default value is false.
+
+* `max_retry_times` - The maximum number of retries.
+  The maximum value is `100`, and the default value is `20`.
+
+* `jars` - The name of the package that is of the JAR type and has been uploaded to the DLI
+  resource management system. You can also specify an OBS path, for example, obs://Bucket name/Package name.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `jars` - The name of the package that is of the JAR type and has been uploaded to the DLI
  resource management system. You can also specify an OBS path, for example, `obs://{bucket name}/{package name}`.
```
check the other places

File: docs/data-sources/dli_spark_templates.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+* `dependent_packages` - The list of package resource objects.
+
+  The [dependent_packages](#body_dependent_packages_struct) structure is documented below.
+
+* `configurations` - The configuration items of the DLI Spark.
+
+* `executor_memory` - The executor memory of the Spark application, for example, 2 GB and 2048 MB.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `executor_memory` - The executor memory of the Spark application, for example, `2` GB and `2,048` MB.
```
check the other places.

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_spark_templates_test.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+}
+  
+locals {
+  template_id = huaweicloud_dli_spark_template.test.id
+}
+  
+output "template_id_filter_is_useful" {
+  value = length(data.huaweicloud_dli_spark_templates.template_id_filter.templates) > 0 && alltrue(
+    [for v in data.huaweicloud_dli_spark_templates.template_id_filter.templates[*].id : v == 
+  local.template_id]
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_spark_templates_test.go]:
```suggestion
    [for v in data.huaweicloud_dli_spark_templates.template_id_filter.templates[*].id : v == local.template_id]
```
maximum 120 characters allowed for a line.
check the other places.

File: docs/data-sources/dli_spark_templates.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 3 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 4 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 5 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_spark_templates_test.go (hunk 6 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_spark_templates_test.go]: done

File: huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+func DataSourceDliSparkTemplates() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDliSparkTemplatesRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the region in which to query the resource. If omitted, the provider-level region will be used.`,
```
Comment (Lance52259) [inline huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go]:
```suggestion
				Description: `Specifies the region in which to query the resource.`,
```

Comment (wangyuancheng01) [reply huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go]: done

**Pull request merged**